test/e2e/e2e_test.go

```diff
@@ -0,0 +1,111 @@
+				GenerateName: "e2e-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+	})
+	ginkgo.AfterEach(func() {
+		gomega.Expect(k8sClient.Delete(ctx, ns)).To(gomega.Succeed())
+	})
+	ginkgo.When("Creating a Job Without Queueing", func() {
+		ginkgo.It("Should stay in suspended", func() {
+			sampleJob := testing.MakeJob("test-job", ns.Name)
+			annotations := map[string]string{
+				"kueue.x-k8s.io/queue-name": "main",
+			}
+			sampleJob.ObjectMeta.Annotations = annotations
+			gomega.Expect(k8sClient.Create(ctx, &sampleJob.Job)).Should(gomega.Succeed())
+			lookupKey := types.NamespacedName{Name: "test-job", Namespace: ns.Name}
+			createdJob := &batchv1.Job{}
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, lookupKey, createdJob); err != nil {
+					return false
+				}
+				return *createdJob.Spec.Suspend
+			}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
+			gomega.Expect(k8sClient.Delete(ctx, &sampleJob.Job)).Should(gomega.Succeed())
+		})
+	})
+	ginkgo.When("Creating A Job With Queueing", func() {
+		var (
+			resourceKueue *kueue.ResourceFlavor
+			localQueue    *kueue.LocalQueue
+			clusterQueue  *kueue.ClusterQueue
+		)
+		ginkgo.BeforeEach(func() {
+			resourceKueue = testing.MakeResourceFlavor("default").Obj()
+			gomega.Expect(k8sClient.Create(ctx, resourceKueue)).Should(gomega.Succeed())
+			localQueue = testing.MakeLocalQueue("main", ns.Name).Obj()
+			clusterQueue = testing.MakeClusterQueue("cluster-queue").
+				Resource(testing.MakeResource(corev1.ResourceCPU).
+					Flavor(testing.MakeFlavor("default", "1").Obj()).Obj()).
+				Resource(testing.MakeResource(corev1.ResourceMemory).
+					Flavor(testing.MakeFlavor("default", "36Gi").Obj()).Obj()).Obj()
+			localQueue.Spec.ClusterQueue = "cluster-queue"
+
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).Should(gomega.Succeed())
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(k8sClient.Delete(ctx, localQueue)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Delete(ctx, clusterQueue)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Delete(ctx, resourceKueue)).Should(gomega.Succeed())
+		})
+		ginkgo.It("Should unsuspend a job", func() {
+			sampleJob := testing.MakeJob("test-job", ns.Name).Request("cpu", "1").Request("memory", "20Mi")
+			annotations := map[string]string{
+				"kueue.x-k8s.io/queue-name": "main",
+			}
+			sampleJob.ObjectMeta.Annotations = annotations
+			gomega.Expect(k8sClient.Create(ctx, &sampleJob.Job)).Should(gomega.Succeed())
+
+			lookupKey := types.NamespacedName{Name: "test-job", Namespace: ns.Name}
+			createdJob := &batchv1.Job{}
+			gomega.Eventually(func() bool {
+				if err := k8sClient.Get(ctx, lookupKey, createdJob); err != nil {
+					return false
+				}
+				return !*createdJob.Spec.Suspend
+			}, framework.Timeout, framework.Interval).Should(gomega.BeTrue())
```

Comment:
Also worth checking that the Job eventually finishes. You can use the image that we use here https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/testing/batch/job.yaml

Reply:
Done.